widgets/passwordrevealer/org.eclipse.nebula.widgets.passwordrevealer/src/org/eclipse/nebula/widgets/passwordrevealer/PasswordRevealer.java

```diff
@@ -0,0 +1,2357 @@
+ * @see <a href="http://www.eclipse.org/swt/">Sample code and further information</a>
+ * @noextend This class is not intended to be subclassed by clients.
+ */
+public class PasswordRevealer extends Composite {
+
+	protected Text passwordField;
+	private final EyeButton eyeButton;
+	private final Text textField;
+	private final Composite comp;
+	private Point currentCaretPosition;
+
+	/**
+	 * Constructs a new instance of this class given its parent and a style value
+	 * describing its behavior and appearance.
+	 * <p>
+	 * The style value is either one of the style constants defined in class
+	 * <code>SWT</code> which is applicable to instances of this class, or must be
+	 * built by <em>bitwise OR</em>'ing together (that is, using the
+	 * <code>int</code> "|" operator) two or more of those <code>SWT</code> style
+	 * constants. The class description lists the style constants that are
+	 * applicable to the class. Style bits are also inherited from superclasses.
+	 * </p>
+	 *
+	 * @param parent a composite control which will be the parent of the new
+	 *            instance (cannot be null)
+	 * @param style the style of control to construct
+	 *
+	 * @exception IllegalArgumentException
+	 *                <ul>
+	 *                <li>ERROR_NULL_ARGUMENT - if the parent is null</li>
+	 *                </ul>
+	 * @exception SWTException
+	 *                <ul>
+	 *                <li>ERROR_THREAD_INVALID_ACCESS - if not called from the
+	 *                thread that created the parent</li>
+	 *                <li>ERROR_INVALID_SUBCLASS - if this class is not an allowed
+	 *                subclass</li>
+	 *                </ul>
+	 *
+	 * @see Widget#getStyle()
+	 */
+	public PasswordRevealer(Composite parent, int style) {
+		super(parent, SWT.BORDER);
+		final GridLayout gl = new GridLayout(2, false);
+		gl.horizontalSpacing = gl.verticalSpacing = gl.marginHeight = gl.marginWidth = 0;
+		setLayout(gl);
+
+		comp = new Composite(this, SWT.NONE);
+		final GridLayout glComp = new GridLayout(1, false);
+		glComp.horizontalSpacing = glComp.verticalSpacing = glComp.marginHeight = glComp.marginWidth = 0;
+		comp.setLayout(glComp);
+		comp.setLayoutData(new GridData(GridData.FILL, GridData.CENTER, true, true));
+
+		passwordField = new Text(comp, style | SWT.PASSWORD | removeFields(style, SWT.BORDER));
+		passwordField.setLayoutData(new GridData(GridData.FILL, GridData.FILL, true, false));
+
+		textField = new Text(comp, style | removeFields(style, SWT.PASSWORD, SWT.BORDER));
+		final GridData gdText = new GridData(GridData.FILL, GridData.FILL, true, false);
+		textField.setLayoutData(gdText);
+		textField.setVisible(false);
+		gdText.exclude = true;
+
+		eyeButton = new EyeButton(this, SWT.NONE);
+		eyeButton.setLayoutData(new GridData(GridData.FILL, GridData.FILL, false, false));
+
+		passwordField.addListener(SWT.Modify, e -> {
+			final String currentText = passwordField.getText();
+			textField.setText(currentText);

```

Comment:
Is it an idea to only field textField when the eye button is pressed and then clear the textField when eye is depressed?

Reply:
Excellent idea.